projects/vdk-core/src/vdk/internal/builtin_plugins/connection/managed_connection_base.py

```diff
@@ -96,9 +96,12 @@ def method(*args, **kwargs):
     )
     def connect(self) -> PEP249Connection:
         """
+        Checks if we are still properly connected to the database (including by issuing validation query)
+        and re-connect if not.
         :return: PEP249 Connection object (managed)
         """
-        if not self._is_db_con_open:
+
+        if not self._is_connected():
```

Comment:
there is no functional change with that modification, since:
```
def _is_connected(self) -> bool:
        if None is self._is_db_con_open:
            return False
        if False is self._is_db_con_open:
            return False
```